src/modules/iotjs_module_crypto.c

```diff
@@ -371,24 +401,95 @@ JS_FUNCTION(Sha1Encode) {
     return jerry_create_undefined();
   }
 
-  unsigned char *sha1_ret = NULL;
-  size_t sha1_sz =
-      iotjs_sha1_encode(&sha1_ret,
-                        (const unsigned char *)iotjs_string_data(&user_str),
-                        iotjs_string_size(&user_str));
+  size_t sha_sz = 0;
+
+  unsigned char *sha_ret = NULL;
+
+  switch (type) {
+    case IOTJS_CRYPTO_SHA1: {
+      sha_sz =
+          iotjs_sha1_encode(&sha_ret,
+                            (const unsigned char *)iotjs_string_data(&user_str),
+                            iotjs_string_size(&user_str));
```

Comment:
The `iotjs_string_data` and `iotjs_string_size` calls can be extracted to a local var before the switch. That way we can reduce the code duplication between the two cases.